client/types.go

```diff
@@ -0,0 +1,155 @@
+package client
+
+import (
+	"strings"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/auth"
+)
+
+// BaseReq defines a structure that can be embedded in other request structures
+// that all share common "base" fields.
+type BaseReq struct {
+	From          string       `json:"from"`
+	Password      string       `json:"password"`
+	Memo          string       `json:"memo"`
+	ChainID       string       `json:"chain_id"`
+	AccountNumber uint64       `json:"account_number"`
+	Sequence      uint64       `json:"sequence"`
+	Fees          sdk.Coins    `json:"fees"`
+	GasPrices     sdk.DecCoins `json:"gas_prices"`
+	Gas           string       `json:"gas"`
+	GasAdjustment string       `json:"gas_adjustment"`
+	GenerateOnly  bool         `json:"generate_only"`
+	Simulate      bool         `json:"simulate"`
+}
+
+// SendReq requests sending an amount of coins
+type SendReq struct {
+	Amount  sdk.Coins `json:"amount"`
+	BaseReq BaseReq   `json:"base_req"`
+}
+
+// MsgDelegationsInput requests a delegation
+type MsgDelegationsInput struct {
```

Comment:
I think we might want to move these to their respective modules, but this can be a separate PR. What do you think @alexanderbez and @alessio?

Reply:
Yes, I noticed this. Some of these types were spread around `client` because they were used for testing. I noticed that in some cases, json is still constructed in strings.
Maybe in another PR, these types could be moved and we can remove some "manual" json strings too.